### PR TITLE
chore: clean up recordings

### PR DESCRIPTION
**Pull request by akshaysngupta**
---
* Added record processor
* You can validate the recording for this test to see if the record processor is working: `src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_create_with_ingress_appgw_addon.yaml`

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
chore

**Comment by akshaysngupta** [comment]
@yungezz Can you help merge this PR first to fix the recordings. This is needed so that the tests succeed.

https://github.com/Azure/azure-cli-extensions/pull/2803

**Comment by yungezz** [comment]
> @yungezz Can you help merge this PR first to fix the recordings. This is needed so that the tests succeed.
> 
> #2803

@houk-ms is helping on fixing CI to get CI passed

**Comment by houk-ms** [comment]
CI failed because Network SDK has upgraded to 2020-07-01, while the recording file is still in 2020-06-01.

For the standard process, we should:
Pull the latest CLI/dev codes and rerun live tests.

Consider the case of credential leaks, you could do a quick fix:
Run recording tests locally, find all the places where the api version of Network mismatches and modify the recording file to 2020-07-01 directly.

**Comment by jsntcy** [comment]
@akshaysngupta , you can change the version for network api to 2020-07-01.

**Comment by qwordy** [comment]
Is it a real secret? If no, you can also add files to scripts/ci/credscan/CredScanSuppressions.json.
If you want to replace the secret, the recommended way is writing your own replacer. E.g.
```
class KeyReplacer(RecordingProcessor):

    def process_request(self, request):
        if is_text_payload(request) and isinstance(request.body, bytes):
            request.body = self._replace_byte_keys(request.body)
        elif is_text_payload(request) and isinstance(request.body, str):
            request.body = self._replace_string_keys(request.body)
        return request

    def process_response(self, response):
        if is_text_payload(response) and response['body']['string']:
            response['body']['string'] = self._replace_string_keys(response['body']['string'])
        return response

    # pylint: disable=no-self-use
    def _replace_string_keys(self, val):
        import re
        if 'primaryKey' in val:
            val = re.sub(r'"primaryKey":( ?)"([^"]+)"', r'"primaryKey":"{}"'
                         .format(MOCK_KEY), val, flags=re.IGNORECASE)
        if 'secondaryKey' in val:
            val = re.sub(r'"secondaryKey":( ?)"([^"]+)"', r'"secondaryKey":"{}"'
                         .format(MOCK_KEY), val, flags=re.IGNORECASE)
        return val

    # pylint: disable=no-self-use
    def _replace_byte_keys(self, val):
        import re
        if b'primaryKey' in val:
            val = re.sub(b'"primaryKey":( ?)"([^"]+)"', '"primaryKey":"{}"'
                         .format(MOCK_KEY).encode(), val, flags=re.IGNORECASE)
        if b'secondaryKey' in val:
            val = re.sub(b'"secondaryKey":( ?)"([^"]+)"', '"secondaryKey":"{}"'
                         .format(MOCK_KEY).encode(), val, flags=re.IGNORECASE)
        return val
```

**Review by houk-ms — APPROVED** [review]

**Comment by feiskyer** [comment]
@akshaysngupta could you add the replacer as suggested by @qwordy above? With the replacer, we could avoid recording credentials in the future.

**Comment by akshaysngupta** [comment]
@houk-ms @qwordy  @feiskyer added record processor.

**Comment by akshaysngupta** [comment]
This also needs to be added to Azure CLI.

**Comment by qwordy** [comment]
Hi @akshaysngupta ,Thank you for adding this record processor. It is still "yyyy" in yaml files.. Should it be replaced with "fake-secret"?

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -165,7 +171,7 @@ def test_aks_create_with_ingress_appgw_addon(self, resource_group, resource_grou
             'name': aks_name
         })
 
-        create_cmd = 'aks create --resource-group={resource_group} --name={name} --enable-managed-identity --service-principal xxxx --client-secret yyyy --generate-ssh-keys ' \
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --enable-managed-identity --generate-ssh-keys ' \
```
Comment (houk-ms) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:174]: why the `--service-principal` and `--client-secret` be removed? this command will fail when running live test

**Review by houk-ms — COMMENTED** [review]
What about the live test when we pass `xxxx` to --service-principal?

**Review by yungezz — APPROVED** [review]

**Review by qwordy — APPROVED** [review]

**Pull request merged**